src/useStyles.js

```diff
@@ -0,0 +1,79 @@
+import { useContext, useRef } from 'react';
+import { DIRECTIONS } from 'react-with-direction';
+
+import withPerf from './utils/perf';
+import WithStylesContext from './WithStylesContext';
+import { _getInterface, _getTheme } from './ThemedStyleSheet';
+
+/**
+ * Hook used to derive the react-with-styles props from the provided react-with-styles
+ * theme, interface, direction, and styles function.
+ *
+ * @export
+ * @param {*} [{ direction, stylesFn, flushBefore }={}]
+ * @returns {*} { css, styles, theme }
+ */
+export default function useStyles({ direction, stylesFn, flushBefore } = {}) {
+  if (!useContext || !useRef) {
+    throw new ReferenceError('useStyles() requires React 16.8 or later');
+  }
+
+  // Get the styles interface and styles theme from context
+  let { stylesInterface, stylesTheme: theme } = useContext(WithStylesContext);
+
+  // Fallback to the singleton implementation
+  stylesInterface = stylesInterface || _getInterface();
```

Comment:
does `_getInterface()` make it available for future useContext calls, or should this somehow be persisted?

Reply:
No it doesn't... I see what you mean. 🤔